graphics/blit-atari.cpp

```diff
@@ -86,13 +112,139 @@ void copyBlit(byte *dst, const byte *src,
 
 		// wait until we finish otherwise we may overwrite pixels written manually afterwards
 		while (*SV_BLITTER_CONTROL & 1);
-	} else if (dstPitch == srcPitch && ((w * bytesPerPixel) == dstPitch)) {
-		memcpy(dst, src, dstPitch * h);
+	} else if (dstPitch == srcPitch) {
```

Comment:
This doesn't work if w is smaller than the whole line.

Reply:
Good catch! I've overlooked that `src` and/or `dst` are generic pointers, not first bytes of a surface.

Reply:
It's not only that.
You sometimes want to blit only a part of the screen.
In this case you set w to the sub part width and pitch to the real width*pixel size.

Reply:
Even better. I'm glad I just didn't merge this, it would have bite me hard later. Thanks.